Update connector with service class inclusion

sushie1984 opened the pull request:
Adds service class inclusion approach by extending any class with `GraphqlConnector::<Name>::Query` in order to build re-usable query methods. See `README` for more details

### Where to start reviewing
[`/queryable.rb`](https://github.com/Garllon/graphql_connector/pull/14/files#diff-898af31520811ea81785cac84770430b)

### Manual testing
* Create one or multiple service class inclusions as described in the `README`
* Perform queries against one or multiple graphql server
* Expect to receive response as if using `query`/`raw_query` of the previously implemented Server approach

Garllon commented on README.md:
```diff
@@ -76,6 +86,94 @@ Example:
+end
+
+Product.all
+=> [ { id: '1', name: 'Demo Product', ...]
```
I think it should be like this, right:
```ruby
[ { id: '1', name: 'Demo Product' }, ...]
```
Otherwise i would expect the `...` means there are more attributes of one object.

sushie1984 replied:
Fixed typo

Garllon commented on lib/graphql_connector/service_classable/query.rb:
```diff
@@ -0,0 +1,104 @@
+
+module GraphqlConnector
+  module ServiceClassable
+    class InvalidTypeMappingError < StandardError
```
I prefer the following syntax:
```ruby
class InvalidTypeMappingError < StandardError; end
```

sushie1984 replied:
Agree that looks much nicer. Refactored it accordingly

Garllon commented on lib/graphql_connector/service_classable/query.rb:
```diff
@@ -0,0 +1,104 @@
+        def validate_mapping!(type_mapping)
+          return if type_mapping.size == 1 &&
+                    type_mapping.first[0].is_a?(Symbol) &&
+                    (type_mapping.first[1].is_a?(Symbol) ||
```
What do you think about:
```ruby
[String, Symbol].member?(type_mapping.first[1].class)
```

And second thing does it make sense to support string a symbols for the second parameter, but not for the first one?
```ruby
'test' => 'blubber'
```

sushie1984 replied:
> What do you think about:
> [String, Symbol].member?(type_mapping.first[1].class)

Yes that looks more readable. Changed it to that

> And second thing does it make sense to support string a symbols for the second parameter, but not for the first one?
> 'test' => 'blubber'

That is a good question. However we cannot use this syntax if want to keep the keyword argument with splat parameters in [`def query(**type_mapping`)](https://github.com/Garllon/graphql_connector/pull/14/files#diff-d14d48c082efaf086801f55333ed8758R26)

sushie1984 replied:
I actually refactored the whole procedure that `String` is also covered (as in your example)

sushie1984 replied:
Further refactorings:

* Renamed `query` to `add_query` (feels more accurate)
* Added `add_raw_query` which deals with raw graphql queries
* Moved `return_fields` into actual `add_query` segment since such a service class can have multiple `graphql` types and therefore a global `return_fields` would have made no sense

sushie1984 commented on lib/graphql_connector.rb:
```diff
@@ -21,6 +22,7 @@ def self.configuration
   end
 
   def self.reset
+    @configuration.reset!
```
Yes that makes sense. Removed obsolete re-assignment

sushie1984 commented:
As talked before -->
* Updated wording in README
* Added examples
* Bugfixed `raw_query` issue (discovered while preparing examples)

See also [here](https://github.com/Garllon/graphql_connector/pull/14/commits/f4d6c7880ff1b6ef5772f992cef9f67c391b3683) and [here](https://github.com/Garllon/graphql_connector/pull/14/commits/2e539087069b17229df9c031aeab75be0281cb6e)

Garllon commented on CHANGELOG.md:
```diff
@@ -1,3 +1,12 @@
+## 1.1.0 (2020-2-8)
```
Can you update the date to today? Please

sushie1984 replied:
Yes that makes totally sense. Adjusted it accordingly
